python/cudf/cudf/core/column/column.py

```diff
@@ -1096,8 +1096,6 @@ def build_categorical_column(
     ordered : bool
         Indicates whether the categories are ordered
     """
-    if len(categories) == 0 and len(codes):
```

Comment:
This would be incorrect when someone wants to construct a categorical column with all nulls and no categories. This is a valid scenario, hence removed this validation.